### PR TITLE
Allow array returns when `skip_vectorization` is being used, enforce import convention.

### PR DESCRIPTION
**Pull request by hmgaudecker**
### What problem do you want to solve?

- Extract the `float` bit from typing info like `numpy.ndarray[float]`, so we can use proper return types.
- Add array types in some cases
- Enforce convention for importing numpy:
  - `np` means our own `numpy_or_jax`
  - `numpy` means numpy proper

### CODE REVIEW

File: src/_gettsim/interface.py [hunk 1]
```diff
@@ -384,23 +391,31 @@ class FunctionsAndColumnsOverlapWarning(UserWarning):
...
+            second_part = format_errors_and_warnings(
+                """
+                These are already present among the hard-coded functions of the taxes
+                and transfers system. If you want a data column to be used instead of
+                calculating it within GETTSIM you do not need to do anything. If you
+                want data columns to be calculated by hard-coded functions, remove them
+                from the *data* you pass to GETTSIM. You need to pick one option for
+                each column that appears in the list above.
+                """
+            )
```
Comment (lars-reimann) [inline src/_gettsim/interface.py:417]: Is this duplication really worth it for correct pluralization? Always using the plural would not hurt readability of the error IMO.

**Review by lars-reimann — APPROVED** [review]
Looks great overall, just one question.

File: src/_gettsim/interface.py (hunk 1 above)
Comment (hmgaudecker) [reply src/_gettsim/interface.py:417]: I think so. When newcomers are lost (and this is a very common point to be confused), I think any little bit of showing that devs took care helps. We had it all along, too, just a different formatting now.

**Review by MImmesberger — COMMENTED** [review]
Shouldn't we add support for the `numpy.ndarray[float]` return type here as well? We encountered that in #751. 

In my commit, I changed the return type from `float` to `numpy.ndarray[float]` for that function. If you don't want to tackle this right now, feel free to revert.

**Comment by hmgaudecker** [comment]
> Shouldn't we add support for the `numpy.ndarray[float]` return type here as well? We encountered that in #751.

Sure, can't believe I missed that one! 

This shows that we'll need to update `convert_series_to_internal_type`, too, however...

**Comment by hmgaudecker** [comment]
Great catch!

**Comment by hmgaudecker** [comment]
Consider it a testimony to the quality of your work that I blindly thought you'd have to be right :smile:

**Pull request merged**